src/librustc_trans/trans/debuginfo.rs

```diff
@@ -1156,12 +1156,42 @@ pub fn get_cleanup_debug_loc_for_ast_node<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
         }
     }
 
-    NodeInfo {
+    NodeIdAndSpan {
         id: node_id,
         span: cleanup_span
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum DebugLocation {
```

Comment:
I thought about that too but then decided to do it this way because `NoDebugLoc` appears in many places (basically for all code the compiler generates) and it's much more informative at these call-sites than a simple `None` in an often long list of parameters. I'd rather keep it that way, if you don't mind.

Regarding the namespacing, I also considered that but there too it becomes unwieldy at the call-sites. `NoDebugLoc` is rather short for the amount of information it carries, while `DebugLoc::None` seems to be more heavyweight. But I don't insist to keep it that way. I'll try it out the namespaced variant and do that if it isn't too bad.